### DISCUSSION TITLE
How to display  modal window  on table  list  when I  click  a cloum text

### DISCUSSION THREAD

**Discussion by oz-hjx210** [filamentphp/filament#6050] [Help] [answered]
For example, if I have a product table with multiple comments for each product, how can I click on the product name in the product list and pop up the corresponding comment box, instead of navigating to another page?

**Comment by danharrin** [comment, marked as answer]
Yes, you can pass a table Action object to the action() method of a Column.

**Reply by oz-hjx210** [reply]
Dan  , can you  give  me a  simple  demo code? thanks

**Reply by danharrin** [reply]
```php
->action(Action::make('addComment')
    ->form([Textarea::make('comment')->required])
    ->action(fn (array $data, $record) => $record->comments()->create(['comment' => $data['comment']])))
```

**Reply by oz-hjx210** [reply]
I'm sorry, maybe I wasn't detailed enough in my expression. What I meant was that I want to see the list of comments on a pop up  modal , not add a comment.
 Thank you for your response. My English level is too poor.

**Reply by danharrin** [reply]
You would pass a `->modalContent(fn ($record) => view('your-custom-view', ['record' => $record]))` to the action instead of `form()`. Then render the comments in your custom view

**Reply by oz-hjx210** [reply]
thanks very much dan  ,you are a superman!

**Comment by divdax** [comment]
I would like to achieve the same outcome as described in the topic. Within my table, I have a column containing custom-styled content that I wish to link to open a modal. Currently, the actions are located at the end of the table, but I would prefer to format the column itself and have the content act as a link. Is there a way to utilize something similar to <a wire:click="openModal('view')"> for this purpose?
